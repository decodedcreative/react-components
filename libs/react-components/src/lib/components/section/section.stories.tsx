import { ComponentStory, Meta } from '@storybook/react';
import { Section } from './section';
import { Typography } from '../typography';

export default {
  component: Section,
  title: 'Theme/Section',
  parameters: {
    backgrounds: {
      default: 'Light',
      values: [
        { name: 'Light', value: '#ffffff' },
        { name: 'Dark', value: '#161b2f' },
      ],
    },
  },
} as Meta;

const Template: ComponentStory<typeof Section> = (args) => (
  <Section {...args}>
    <Section.Item>
      <Typography as="h2" size="xxl">
        Section heading
      </Typography>
      <Typography marginBottom>
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industry's standard dummy text ever
        since the 1500s, when an unknown printer took a galley of type and
        scrambled it to make a type specimen book.
      </Typography>
      <Typography marginBottom>
        {' '}
        The point of using Lorem Ipsum is that it has a more-or-less normal
        distribution of letters, as opposed to using 'Content here, content
        here', making it look like readable English. Many desktop publishing
        packages and web page editors now use Lorem Ipsum as their default model
        text, and a search for 'lorem ipsum' will uncover many web sites still
        in their infancy. Various versions have evolved over the years,
        sometimes by accident, sometimes on purpose (injected humour and the
        like).
      </Typography>
    </Section.Item>
  </Section>
);

export const Default = Template.bind({});
Default.args = {};

export const Primary = Template.bind({});
Primary.args = {
  variant: 'primary',
};

export const Secondary = Template.bind({});
Secondary.args = {
  variant: 'secondary',
};
