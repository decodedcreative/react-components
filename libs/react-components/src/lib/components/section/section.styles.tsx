import styled, { css } from 'styled-components';
import { SectionProps, SectionItemProps } from './section';

export const Section = styled('section')<SectionProps>`
  padding-top: ${(props) => props.theme.spacing.lg};
  padding-bottom: ${(props) => props.theme.spacing.lg};

  ${(props) =>
    props.variant === 'primary' &&
    css`
      background-color: ${(props) => props.theme.colors.color20};
    `}

  ${(props) =>
    props.variant === 'secondary' &&
    css`
      background-color: ${(props) => props.theme.colors.color30};
    `}
`;

export const SectionContent = styled('div')<SectionProps>`
  padding-left: ${(props) => props.theme.general.gridGutter};
  padding-right: ${(props) => props.theme.general.gridGutter};
`;

export const SectionItem = styled('div')<SectionItemProps>``;

export const SectionItemContent = styled('div')`
  max-width: ${(props) => props.theme.general.gridMaxWidth};
  margin-left: auto;
  margin-right: auto;
`;
