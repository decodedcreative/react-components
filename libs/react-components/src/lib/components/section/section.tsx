import { ReactNode } from 'react';
import * as S from './section.styles';

export interface SectionProps {
  children: ReactNode;
  variant?: 'primary' | 'secondary';
  as?: React.ElementType | keyof JSX.IntrinsicElements;
}

export interface SectionItemProps {
  children: ReactNode;
}

export const Section = ({ children, ...args }: SectionProps) => (
  <S.Section {...args}>
    <S.SectionContent>{children}</S.SectionContent>
  </S.Section>
);

const SectionItem = ({ children }: SectionItemProps) => (
  <S.SectionItem>
    <S.SectionItemContent>{children}</S.SectionItemContent>
  </S.SectionItem>
);

Section.Item = SectionItem;
