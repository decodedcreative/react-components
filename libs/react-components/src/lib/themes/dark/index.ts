import colors from './colors';
import components from './components';
import general from './general';
import spacing from './spacing';
import typography from './typography';

const darkTheme = {
  name: 'darkTheme',
  colors,
  components,
  general,
  spacing,
  typography,
};

export default darkTheme;
