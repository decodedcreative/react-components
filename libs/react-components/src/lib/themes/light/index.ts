import colors from './colors';
import components from './components';
import general from './general';
import typography from './typography';

const lightTheme = {
  name: 'lightTheme',
  colors,
  components,
  general,
  typography,
};

export default lightTheme;
