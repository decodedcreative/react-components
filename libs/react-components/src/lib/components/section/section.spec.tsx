import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../../themes';
import { Section, SectionProps } from './section';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

const createComponentJSX = ({
  children = 'Section text here',
  as,
}: SectionProps) => (
  <ThemeProvider theme={lightTheme}>
    <Section as={as}>{children}</Section>
  </ThemeProvider>
);

const createComponentRender = ({
  children = 'Section text here',
  as,
}: SectionProps) =>
  render(
    <ThemeProvider theme={lightTheme}>
      <Section as={as}>{children}</Section>
    </ThemeProvider>
  );

describe('Section', () => {
  it('should render successfully', () => {
    const component = createComponentRender({} as SectionProps);
    expect(component).toBeTruthy();
  });

  describe('Section - Variants', () => {
    it('should match the primary section snapshot', () => {
      const primarySection = createComponentJSX({} as SectionProps);
      const primarySectionJSON = renderer.create(primarySection).toJSON();
      expect(primarySectionJSON).toMatchSnapshot();
    });
    it('should match the secondary section snapshot', () => {
      const secondarySection = createComponentJSX({
        variant: 'secondary',
      } as SectionProps);
      const secondarySectionJSON = renderer.create(secondarySection).toJSON();
      expect(secondarySectionJSON).toMatchSnapshot();
    });
  });
});
