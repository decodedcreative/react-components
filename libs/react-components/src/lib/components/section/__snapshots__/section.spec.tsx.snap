// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Section - Variants should match the primary section snapshot 1`] = `
.c0 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.c1 {
  padding-left: 1rem;
  padding-right: 1rem;
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    Section text here
  </div>
</section>
`;

exports[`Section Section - Variants should match the secondary section snapshot 1`] = `
.c0 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.c1 {
  padding-left: 1rem;
  padding-right: 1rem;
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    Section text here
  </div>
</section>
`;
